### PR TITLE
Androapp 5114

### PR DESCRIPTION
**Pull request by CarlosMacaneta**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://jira.dhis2.org/browse/

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [x] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [x] Other
## Checklist
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/screens/components/Toolbar.kt [hunk 1]
```diff
@@ -40,7 +40,7 @@ fun Toolbar(
     from: String,
     to: String?,
     themeColor: Color,
-    navigationAction: () -> Unit,
+    launchBottomSheet: () -> Unit,
```
Comment (andresmr) [inline stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/screens/components/Toolbar.kt:43]: does this always launch the bottom sheet or only if has data entered?. Maybe you can go from here to the viewmodel, check there the status and then decide if go back or launch the bottom sheet. Also, with this you don't need to expose the `canReview()` method from the viewModel. What do you think?

**Review by andresmr — CHANGES_REQUESTED** [review]

File: stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/screens/components/BackdropComponent.kt [hunk 2]
```diff
@@ -57,7 +58,7 @@ fun Backdrop(
                 settingsUiState.deliverToLabel()?.asString(),
                 themeColor,
                 launchBottomSheet = {
-                    if (manageStockViewModel.canReview()) {
+                    if (manageStockViewModel.dataEntryUiState.value.hasUnsavedData) {
```
Comment (andresmr) [inline stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/screens/components/BackdropComponent.kt:61]: you can use `dataEntryUiState.hasUnsavedData` property

File: stock-managment/src/main/java/org/dhis2/android/rtsm/ui/managestock/ManageStockViewModel.kt [hunk 3]
```diff
@@ -350,7 +352,7 @@ class ManageStockViewModel @Inject constructor(
         }
 
         _dataEntryUiState.update { currentUiState ->
-            currentUiState.copy(button = button)
+            currentUiState.copy(button = button, hasUnsavedData = hasUnsavedData)
```
Comment (andresmr) [inline stock-managment/src/main/java/org/dhis2/android/rtsm/ui/managestock/ManageStockViewModel.kt:355]: when is unsavedData set to false?

File: stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/screens/components/BackdropComponent.kt (hunk 2 above)
Comment (CarlosMacaneta) [reply stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/screens/components/BackdropComponent.kt:61]: yes, indeed

**Review by andresmr — APPROVED** [review]

**Pull request merged**